book/src/01-04-clearing-the-screen.md

```diff
@@ -0,0 +1,153 @@
+# Clearing the Screen
+
+In this section, you'll learn how to clear the screen and handle window resizes.
+
+## The swap chain
+
+`wgpu` represents the framebuffers that will be presented to the screen with a data structure called a `SwapChain`.
+If you're coming from Vulkan or D3D12, you'll already be familiar with this term.
+Briefly, the `SwapChain` determines the format, size and presentation mode of the frames drawn by `wgpu`.
+
+A `SwapChain` is created using `Device::create_swap_chain`:
+
+```rust,no_run,no_playground
+async fn run(event_loop: EventLoop<()>, window: Window) {
+    // let (device, queue) = ...
+    
+    let swapchain_format = wgpu::TextureFormat::Bgra8UnormSrgb;
+
+    let size = window.inner_size();
+    let sc_desc = wgpu::SwapChainDescriptor {
+        usage: wgpu::TextureUsage::OUTPUT_ATTACHMENT,
+        format: swapchain_format,
+        width: size.width,
+        height: size.height,
+        present_mode: wgpu::PresentMode::Mailbox,
+    };
+
+    let mut swap_chain = device.create_swap_chain(&surface, &sc_desc);
+
+    // event_loop.run(...);
+}
+```
+
+A few notes on this code:
+- The `width` and `height` fields of `SwapChainDescriptor` take their values from the window size.
+  This is not maintained automatically, so the swap chain needs to be recreated when the window size changes.
+- The choice of `format` is platform-dependent. `Bgra8UnormSrgb` is ubiquitous on desktop Vulkan implementations ([Windows], [Linux]), [required on D3D12], and [the default for Metal] regardless of platform. When developing for mobile or the web, a different format may be required.
```

Comment:
we actually have `swap_chain_get_preferred_format` in wgpu-core. We should expose this in `wgpu-rs`